src/core/api/thread_api.cpp

```diff
@@ -39,6 +39,7 @@
 #include "common/instance.hpp"
 #include "common/locator-getters.hpp"
 #include "common/settings.hpp"
+#include "common/utf8.hpp"
```

Comment:
move to core/utils?

Reply:
How about adding the `ValidateUtf8` to the existing `string.hpp` (It contains a bunch other string related helper methods/functions).

Reply:
Moved to `string.hpp`.